scalalib/test/src/mill/scalalib/CoursierMirrorTests.scala

```diff
@@ -29,14 +30,48 @@ object CoursierMirrorTests extends TestSuite {
     t(eval)
   }
 
+  // The below tests pass individually. Setting up the proxy to an
+  // invalid proxy and setting system properties has side effects that
+  // will break other tests in the test suite, so I recommend you only
+  // uncomment when testing config integration locally.
+
   def tests: Tests = Tests {
-    sys.props("coursier.mirrors") = (resourcePath / "mirror.properties").toString
-    "readMirror" - workspaceTest(CoursierTest) { eval =>
+    "readMirror - system prop" - workspaceTest(CoursierTest) { eval =>
+      sys.props("coursier.mirrors") = (resourcePath / "mirror.properties").toString()
       val Right((result, evalCount)) = eval.apply(CoursierTest.core.repositoriesTask)
       val centralReplaced = result.exists { repo =>
         repo.repr.contains("https://repo.maven.apache.org/maven2")
       }
       assert(centralReplaced)
     }
+
+    // we can't use sys.props to isolate these. It will use the first
+    // value of any prop set in the props in any test. This will
+    // persist beyond the lifetime of the test to the other tests in
+    // the task run. I believe we would need to fork the jvm for each
+    // test in order to achieve isolation, but this would be
+    // costly. If anyone has ideas of how to better test these in
+    // isolation, please provide a PR.
```

Comment:
This is better tested in an integration test. We already have integration tests setup under `integration`.